configs/ote/custom-counting-instance-seg/resnet50_maskrcnn/model.py

```diff
@@ -0,0 +1,177 @@
+            bbox_coder=dict(type='DeltaXYWHBBoxCoder',
+                            target_means=[0.0, 0.0, 0.0, 0.0],
+                            target_stds=[0.1, 0.1, 0.2, 0.2]),
+            reg_class_agnostic=False,
+            loss_cls=dict(type='CrossEntropyLoss',
+                          use_sigmoid=False,
+                          loss_weight=1.0),
+            loss_bbox=dict(type='L1Loss', loss_weight=1.0)),
+        mask_roi_extractor=dict(
+            type='SingleRoIExtractor',
+            roi_layer=dict(type='RoIAlign',
+                           output_size=14,
+                           sampling_ratio=0),
+            out_channels=256,
+            featmap_strides=[4, 8, 16, 32]),
+        mask_head=dict(
+            type='FCNMaskHead',
+            num_convs=4,
+            in_channels=256,
+            conv_out_channels=256,
+            num_classes=80,
+            loss_mask=dict(type='CrossEntropyLoss',
+                           use_mask=True,
+                           loss_weight=1.0))),
+    train_cfg=dict(
+        rpn=dict(
+            assigner=dict(
+                type='MaxIoUAssigner',
+                pos_iou_thr=0.7,
+                neg_iou_thr=0.3,
+                min_pos_iou=0.3,
+                match_low_quality=True,
+                ignore_iof_thr=-1,
+                gpu_assign_thr=300),
+            sampler=dict(
+                type='RandomSampler',
+                num=256,
+                pos_fraction=0.5,
+                neg_pos_ub=-1,
+                add_gt_as_proposals=False),
+            allowed_border=-1,
+            pos_weight=-1,
+            debug=False),
+        rpn_proposal=dict(
+            nms_across_levels=False,
+            nms_pre=2000,
+            nms_post=1000,
+            max_num=1000,
+            nms_thr=0.7,
+            min_bbox_size=0),
+        rcnn=dict(
+            assigner=dict(
+                type='MaxIoUAssigner',
+                pos_iou_thr=0.5,
+                neg_iou_thr=0.5,
+                min_pos_iou=0.5,
+                match_low_quality=True,
+                ignore_iof_thr=-1,
+                gpu_assign_thr=300),
+            sampler=dict(
+                type='RandomSampler',
+                num=512,
+                pos_fraction=0.25,
+                neg_pos_ub=-1,
+                add_gt_as_proposals=True),
+            mask_size=28,
+            pos_weight=-1,
+            debug=False)),
+    test_cfg=dict(
+        rpn=dict(
+            nms_across_levels=False,
+            nms_pre=1000,
+            nms_post=1000,
+            max_num=1000,
+            nms_thr=0.7,
+            max_per_img=1000,
+            min_bbox_size=0),
+        rcnn=dict(
+            score_thr=0.05,
+            nms=dict(
+                type='nms', iou_threshold=0.5, max_num=100),
+            max_per_img=100,
+            mask_thr_binary=0.5)))
+
+cudnn_benchmark = True
+evaluation = dict(interval=1, metric='mIoU', save_best='mIoU')
```

Comment:
Why did you choose mIoU but not segm AP?

Reply:
I'm not sure about using mAP0.5 as it's not representative of picking the most accurate model (50% overlap of GT mask and detect mask I think it's too loose to count as TP mask). 

In my observation selecting a model based on mIoU gives better masks. We can also use mAP0.5:0.95 which I think gives more or less the same model

Reply:
I would use mAP0.5:0.95, but maybe it can be added in next PR

Reply:
Do we have out-of-the-box support for mAP0.5:0.95 in the custom dataset that we use? If yes, and from algo stand point you see no clear evidence that it's beneficial to use mIoU, I'd suggest going with the standard functionality rather than adding an ad hoc one. If no, and we have to add support for something anyway, then it's up to you what to add here.

Reply:
@druzhkov-paul there's no out-of-box mAP0.5:0.95 support for mask mAP in mmdetection, so I added a function `mmdet/core/evaluation/mean_ap_seg.py::eval_segm` that supports both mAP0.5:0.95 and mIoU. Thus, we can switch to mAP0.5:0.95 if we prefer, and personally, I'm fine with either.

Reply:
let's use mAP0.5:0.95

Reply:
updated